src/components/Leaderboards.astro

```diff
@@ -0,0 +1,22 @@
+---
+import { getAuth } from "firebase-admin/auth";
+import { app } from "../lib/firebase_server.ts";
+import LeaderboardServer from "./ServerLeaderboard.astro";
+import LeaderboardClient from "./ClientLeaderboard.astro";
+
+let loggedIn = false;
+let session = undefined;
+const auth = getAuth(app);
+if (Astro.cookies.has("__session")) {
+  const sessionCookie = Astro.cookies.get("__session")?.value;
+  if (sessionCookie) {
+    try {
+      const decodedCookie = await auth.verifySessionCookie(sessionCookie);
+      session = sessionCookie;
+      loggedIn = true;
+    } catch {}
+  }
+}
+---
+
+{ loggedIn ? <LeaderboardServer/> : <LeaderboardClient /> }
```

Comment:
Does this component actually need to have different behavior based on whether or not you're logged in?